moveit_core/robot_trajectory/include/moveit/robot_trajectory/robot_trajectory.h

```diff
@@ -51,8 +52,15 @@ MOVEIT_CLASS_FORWARD(RobotTrajectory);  // Defines RobotTrajectoryPtr, ConstPtr,
 class RobotTrajectory
 {
 public:
+  /** @brief construct a trajectory */
+  explicit RobotTrajectory(const moveit::core::RobotModelConstPtr& robot_model);
+
+  /** @brief construct a trajectory for the named JointModelGroup */
   RobotTrajectory(const moveit::core::RobotModelConstPtr& robot_model, const std::string& group);
 
+  /** @brief construct a trajectory for the JointModelGroup
+   *  If group is nullptr this is equivalent to the first constructor
```

Comment:
Sorry, late to the party, could you please explain (in another PR) in the comments what it actually means for a `RobotTrajectory` to have a group set (or what happens when you use the default constructor)

:+1: for starting with the comments!